streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -174,7 +174,11 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                 final ConsumerRecords<byte[], byte[]> records = consumer.poll(100);
                 for (ConsumerRecord<byte[], byte[]> record : records) {
                     offset = record.offset() + 1;
-                    stateRestoreCallback.restore(record.key(), record.value());
+                    if (record.key() == null) {
+                        log.warn("null key found at offset %d while restoring state from topic %s and partition %d. Skipping record", record.offset(), record.topic(), record.partition());
```

Comment:
Could a log level of WARN wreak havoc in case there's a lot of null keys?

I think the log level we pick here should (a) be consistent with what we use elsewhere (not implying it isn't in this PR!) and (b) it should match the guarantees we have documented for global tables.  If, for example, we clearly tell the user that null keys are ignored, we could use a less chatty log level than WARN.